LoopKit/InsulinKit/DoseStore.swift

```diff
@@ -271,13 +273,19 @@ public final class DoseStore {
 
     private let persistenceController: PersistenceController
 
+    private let cacheLength: TimeInterval
+
     private var purgeableValuesPredicate: NSPredicate {
         return NSPredicate(format: "date < %@", cacheStartDate as NSDate)
```

Comment:
[nit] Random thought, not a blocker or even requiring a response, just food for thought:
I think we should consider this pattern more often... That is, I think it would be a good idea to collect a bunch of common `NSPredicate` constructors and put them into a helper class, so as to avoid the pitfall of typos in the format (which are only found at runtime).

Reply:
Well, there are a *lot* of things that could be cleaned up across all of the stores. Lots of duplicate (or nearly duplicate) code. Not the time or place, though.

Also, regarding the predicate constructors, unfortunately, there is not a great amount of consistency with Core Data properties names across the stores. (eg. `date` vs. `startDate` vs. `timestamp`)

Reply:
Also, again, this was just a [nit].